tests/regression/test_fs_caching.py

```diff
@@ -9,7 +9,7 @@ def howmany(cls):
         try:
             if isinstance(x, cls):
                 n += 1
-        except ReferenceError:
+        except (ReferenceError, AttributeError):
```

Comment:
Can you explain what the cause of the `lru_cache` attribute error was? As far as I remember, we didn't see the error here before refactoring.

Reply:
Some bug with `python3.6/abc.py` at line `183`. 
```python
    def __instancecheck__(cls, instance):
        """Override for isinstance(instance, cls)."""
        # Inline the cache checking
        subclass = instance.__class__
```
where `instance` is `functools._lru_list_elem object`. In the debugger, one happily gets the `__class__` attribute if you break on error in pytest
```
(Pdb++) instance.__class__
<class 'functools._lru_list_elem'>
```
but for some reason it doesn't work.

I must assume that, since this works fine on python 3.7, it's a bug that got fixed.

See commit comments of d3b554d1a6d5a34dd6c8ba7d1e85f6b4cfcf6469

Reply:
Fine.